src/pages/VerifyEmailPage/index.tsx

```diff
@@ -0,0 +1,80 @@
+import { useState, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Box, IconButton } from '@mui/material';
+import ArrowLeftIcon from 'src/assets/icons/arrow-left.svg';
+import StyledButton from 'src/components/shared/StyledButton';
+import {
+  PaddingVariants,
+  StyleVariants,
+} from 'src/components/shared/StyledButton/types';
+import Title from 'src/components/shared/Title';
+import RegularText from 'src/components/shared/RegularText';
+import LabelText from 'src/components/shared/LabelText';
+import TextButton from 'src/components/shared/TextButton';
+import ButtonText from 'src/components/shared/ButtonText';
+
+function VerifyEmailPage() {
+  const timerMin: number = 0;
+  const timerMax: number = 55;
+  const intervalStep: number = 1000;
+  const symbolsForTimer: number = 2;
+
+  const { t } = useTranslation();
+  const [timer, setTimer] = useState<number>(timerMax);
+
+  const formattedTimer = timer.toString().padStart(symbolsForTimer, '0');
+  const isSendAgainButtonDisabled = timer > timerMin;
```

Comment:
I dont like it, it recalculated with each rerender, wrap it infunction and add memoisation